services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -474,34 +484,113 @@ private void doDeleteFlow(final CommandMessage message, String replyToTopic, Des
+            InfoMessage response = new InfoMessage(new FlowRemoveResponse(), System.currentTimeMillis(),
+                    message.getCorrelationId());
+            producerService.sendMessageAndTrack(replyToTopic, message.getCorrelationId(), response);
+
+        } catch (SwitchOperationException e) {
+            logger.error("Failed to process switch rule deletion for switch: '{}'", request.getSwitchId(), e);
+            anError(ErrorType.DELETION_FAILURE)
+                    .withMessage(e.getMessage())
+                    .withDescription(request.getSwitchId().toString())
+                    .withCorrelationId(message.getCorrelationId())
+                    .withTopic(replyToTopic)
+                    .sendVia(producerService);
+        }
     }
 
-    private void processDeleteFlow(RemoveFlow command, DatapathId dpid) throws FlowCommandException {
-        ISwitchManager switchManager = context.getSwitchManager();
+    /**
+     * Reinstall default flow.
+     *
+     * @param message command message for flow deletion
+     */
+    private void doReinstallDefaultFlowForSwitchManager(CommandMessage message) {
+        ReinstallDefaultFlowForSwitchManagerRequest request =
+                (ReinstallDefaultFlowForSwitchManagerRequest) message.getData();
+        IKafkaProducerService producerService = getKafkaProducer();
+        String replyToTopic = context.getKafkaSwitchManagerTopic();
+
+        long cookie = request.getCookie();
+
+        if (!Cookie.isDefaultRule(cookie)) {
+            logger.warn("Failed to reinstall default switch rule for switch: '{}'.  Rule is not default.",
+                    request.getSwitchId());
+            anError(ErrorType.DATA_INVALID)
+                    .withMessage("Failed to reinstall default switch rule. Rule is not default")
+                    .withDescription(request.getSwitchId().toString())
+                    .withCorrelationId(message.getCorrelationId())
+                    .withTopic(replyToTopic)
+                    .sendVia(producerService);
+        }
+
+        SwitchId switchId = request.getSwitchId();
+        DatapathId dpid = DatapathId.of(switchId.toLong());
         try {
-            logger.info("Deleting flow {} from switch {}", command.getId(), dpid);
-
-            DeleteRulesCriteria criteria = Optional.ofNullable(command.getCriteria())
-                    .orElseGet(() -> DeleteRulesCriteria.builder().cookie(command.getCookie()).build());
-            List<Long> cookiesOfRemovedRules = switchManager.deleteRulesByCriteria(dpid, criteria);
-            if (cookiesOfRemovedRules.isEmpty()) {
-                logger.warn("No rules were removed by criteria {} for flow {} from switch {}",
-                        criteria, command.getId(), dpid);
+            RemoveFlow command = new RemoveFlow(null, "REMOVE_DEFAULT_FLOW", cookie, switchId, null, null);
+            List<Long> removedFlows = processDeleteFlow(command, dpid);
```

Comment:
in case of of excess rules this dlete request will return 2 same cookies in this list. Then you will install default rule twice

Reply:
Fixed.